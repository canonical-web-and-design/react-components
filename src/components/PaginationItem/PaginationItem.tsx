import classNames from "classnames";
import PropTypes from "prop-types";
import React, { MouseEventHandler } from "react";

type Props = {
  number: number;
  onClick: MouseEventHandler<HTMLButtonElement>;
  isActive?: boolean;
};

const PaginationItem = ({
  number,
  onClick,
  isActive = false,
}: Props): JSX.Element => (
  <li className="p-pagination__item">
    <button
      className={classNames("p-pagination__link", {
        "is-active": isActive,
      })}
      onClick={onClick}
    >
      {number}
    </button>
  </li>
);

PaginationItem.propTypes = {
  number: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
  isActive: PropTypes.bool,
};

export default PaginationItem;
