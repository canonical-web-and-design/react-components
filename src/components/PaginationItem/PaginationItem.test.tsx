import { shallow } from "enzyme";
import React from "react";

import PaginationItem from "./PaginationItem";

describe("<PaginationItem />", () => {
  // snapshot tests
  it("renders and matches the snapshot", () => {
    const component = shallow(
      <PaginationItem number={1} onClick={jest.fn()} />
    );

    expect(component).toMatchSnapshot();
  });

  // unit tests
  it("displays the page number", () => {
    const component = shallow(
      <PaginationItem number={1} onClick={jest.fn()} />
    );

    expect(component.find("button").text()).toEqual("1");
  });

  it("sets isActive", () => {
    const component = shallow(
      <PaginationItem number={1} onClick={jest.fn()} isActive />
    );

    expect(component.find("button").hasClass("is-active"));
  });
});
