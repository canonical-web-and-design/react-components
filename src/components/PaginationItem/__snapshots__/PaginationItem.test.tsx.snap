// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaginationItem /> renders and matches the snapshot 1`] = `
<li
  className="p-pagination__item"
>
  <button
    className="p-pagination__link"
    onClick={[MockFunction]}
  >
    1
  </button>
</li>
`;
